### PR TITLE
Economic national id (Shenas-e melli)

### PR DESCRIPTION
**Pull request by mabedis**
Hi there,

Firstly, I should say thanks a lot for your valuable & useful project and your clean codes.
Due to lack of an economic national id (shenas-e melli) validator and generator, and despite there are few sources, I could find a [website](http://www.aliarash.com/article/shenasameli/shenasa_meli.htm) that explains how to validate an economic national id, so I tried to implement it according to the structure of your project.

That would be great if you read my code and if everything is fine, accept my pr and add this module to your project.

_Best Regards,
M. A_

### CODE REVIEW

**Comment by mrunderline** [comment]
Hey there!

It's my pleasure to see this project was useful for you, and glad to see your contribution! I've been add some comment on your PR to keep project in a suitable situation.
Feel free to ask any question or any update on your PR.

Best wishes

File: persian_tools/economic_national_id/__init__.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+def generate_random() -> str:
+    """Generate valid Economic National ID according the released validation algorithms.
+
+    Returns:
+        str: A valid economic national ID.
+    """    
+    coef = [29, 27, 23, 19, 17]
+
+    while True:
+        counter += 1
```
Comment (mrunderline) [inline persian_tools/economic_national_id/__init__.py:38]: `counter` variable never defined, how do you change that value?

File: persian_tools/economic_national_id/__init__.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+def generate_random() -> str:
+    """Generate valid Economic National ID according the released validation algorithms.
+
+    Returns:
+        str: A valid economic national ID.
+    """    
+    coef = [29, 27, 23, 19, 17]
+
+    while True:
```
Comment (mrunderline) [inline persian_tools/economic_national_id/__init__.py:37]: It's not too bad, but would you design an algorithm to generate economic_national_id directly and without an `while True` loop?

File: persian_tools/economic_national_id/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class InvalidEconomicNationalId(Exception):
```
Comment (mrunderline) [inline persian_tools/economic_national_id/exceptions.py:1]: Are you using this class anywhere?

File: persian_tools/economic_national_id/__init__.py (hunk 1 above)
Comment (mabedis) [reply persian_tools/economic_national_id/__init__.py:38]: You are right. It was mistake and should be defined before using that.

File: persian_tools/economic_national_id/exceptions.py (hunk 3 above)
Comment (mabedis) [reply persian_tools/economic_national_id/exceptions.py:1]: This class is remained while I was looking for implementing find_place function but there were no information.
I'll get rid of that.

File: persian_tools/economic_national_id/__init__.py (hunk 1 above)
Comment (mrunderline) [reply persian_tools/economic_national_id/__init__.py:38]: please just remove that

File: persian_tools/economic_national_id/__init__.py (hunk 2 above)
Comment (mabedis) [reply persian_tools/economic_national_id/__init__.py:37]: I got it and I found where was my problem.
Thanks. This edit is much better and logical.

**Comment by mrunderline** [comment]
Your PR merged!
thanks for your attention and patient!

**Pull request merged**